src/common/container.h

```diff
@@ -80,61 +81,99 @@ class FixedArray {
   T data_[N];
 };
 
-// A simple fixed size, compile-time allocated deque.
-// Any positiv CAPACITY parameter above 1 should work. Choosing a power of
-// two will typically generate faster code as modulo ops can become bit-AND ops.
-//
-// For a cheap empty() and size() implementation, the actual capacity() is one
-// less than the chosen CAPACITY template parameter.
-template <typename T, int CAPACITY>
-class RingDeque {
- public:
-  RingDeque() {
-    static_assert(CAPACITY > 1, "Capacity needs to be > 1");
-    // intentionally not initializing memory to better see if users do.
+template <typename T, typename... Args>
+std::unique_ptr<T> make_unique(Args &&...args) {
+  return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+}
+
+template <typename T, size_t CAPACITY>
+struct FixedStoragePolicy {
+  static_assert(CAPACITY > 1, "Capacity needs to be > 1");
+  // intentionally not initializing memory to better see if users do.
+  T buffer[CAPACITY];
+  using type = T;
+  T *GetBuffer() { return &buffer[0]; }
+  static constexpr size_t storage_size = CAPACITY;
+};
+
+template <typename T>
+struct DynamicStoragePolicy {
```

Comment:
'Dynamic' is a bad name, as it is not dynamically re-allocating anything. It is still a fixed-size container, so the naming shtould reflect that.

Reply:
(solved) See previous comment